### PR TITLE
Don't reset score when entering/exiting spectator

### PR DESCRIPTION
**Pull request by MrSquid47**

### CODE REVIEW

**Comment by geominorai** [comment]
According to https://wiki.alliedmods.net/Generic_Source_Events#player_team, the `player_team` event should have an `oldteam` value.  If we separate the class and team hook events, should be able to use it and avoid having to check for a fallback value `-1` when it is missing due to getting a `player_class` event.

File: scripting/jse_scoreboard.sp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 bool g_bJSECoreLoaded;
 
 int g_iScore[MAXPLAYERS + 1];
+int g_iLastTeam[MAXPLAYERS + 1];
```
Comment (geominorai) [inline scripting/jse_scoreboard.sp:15]: Possibly unneeded if we can get `oldteam` from the `player_team` event.

File: scripting/jse_scoreboard.sp [hunk 2]
```diff
@@ -90,11 +95,18 @@ public Action CommandListener_Restart(int iClient, const char[] sCommand, int iA
...
 	if (!iClient) {
 		return Plugin_Handled;
 	}
 	
-	ResetClient(iClient);
+	if ((iTeam != 1 && g_iLastTeam[iClient] != iTeam) || iTeam == -1) {
+		ResetClient(iClient);
+	}
+	
+	if (iTeam == 2 || iTeam == 3) {
```
Comment (geominorai) [inline scripting/jse_scoreboard.sp:107]: Preferably convert to TFTeam enums

**Comment by MrSquid47** [comment]
> According to https://wiki.alliedmods.net/Generic_Source_Events#player_team, the `player_team` event should have an `oldteam` value.  If we separate the class and team hook events, should be able to use it and avoid having to check for a fallback value `-1` when it is missing due to getting a `player_class` event.

I don't mind splitting the handlers, I was just keeping the changes minimal.

File: scripting/jse_scoreboard.sp (hunk 1 above)
Comment (MrSquid47) [reply scripting/jse_scoreboard.sp:15]: That would return spectator when returning to a non spec team. The point of this variable is to know if they switched to the opposite team or not.

**Review by geominorai — APPROVED** [review]
Looks good, thanks!

**Pull request merged**